atlasdb-client/src/main/java/com/palantir/atlasdb/util/MetricsManager.java

```diff
@@ -70,11 +74,12 @@ public void registerMetric(Class clazz, String metricName, Gauge gauge) {
     }
 
     public void registerMetric(Class clazz, String metricName, Gauge gauge, Map<String, String> tag) {
-        taggedMetricRegistry.gauge(MetricName.builder()
-                        .safeName(MetricRegistry.name(clazz, metricName))
-                        .safeTags(tag)
-                        .build(),
-                gauge);
+        MetricName taggedMetricName = MetricName.builder()
+                .safeName(MetricRegistry.name(clazz, metricName))
+                .safeTags(tag)
+                .build();
+        taggedMetricRegistry.gauge(taggedMetricName, gauge);
```

Comment:
if there is an existing metric with this name, we should remove it from the registry before adding the new one